drivers/sensor/adxl375/adxl375.c

```diff
@@ -0,0 +1,212 @@
+ * Copyright (c) 2023 Aaron Chan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl375
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <string.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys/__assert.h>
+#include <stdlib.h>
+#include <zephyr/logging/log.h>
+
+#include "adxl375.h"
+
+LOG_MODULE_REGISTER(ADXL375, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl375_check_id(const struct device *dev)
+{
+	struct adxl375_data *data = dev->data;
+	uint8_t device_id = 0;
+
+	int ret = data->hw_tf->read_reg(dev, ADXL375_DEVID, &device_id);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (ADXL375_DEVID_VAL != device_id) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int adxl375_set_odr_and_lp(const struct device *dev, uint32_t data_rate,
+				  const bool low_power)
+{
+	struct adxl375_data *data = dev->data;
+	if (low_power) {
+		data_rate |= 0x8;
+	}
+
+	return data->hw_tf->write_reg(dev, ADXL375_BW_RATE, data_rate);
+}
+
+static int adxl375_set_op_mode(const struct device *dev, enum adxl375_op_mode op_mode)
+{
+	struct adxl375_data *data = dev->data;
+	return data->hw_tf->write_reg(dev, ADXL375_POWER_CTL, op_mode);
+}
+
+static int adxl375_set_data_format(const struct device *dev, uint8_t val)
+{
+	struct adxl375_data *data = dev->data;
+	return data->hw_tf->write_reg(dev, ADXL375_DATA_FORMAT, val);
+}
+
+static int adxl375_init(const struct device *dev)
+{
+	int ret;
+	const struct adxl375_dev_config *cfg = dev->config;
+
+	ret = cfg->bus_init(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize sensor bus.");
+		return ret;
+	}
+
+	ret = adxl375_check_id(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to get valid device ID.");
+		return ret;
+	}
+
+	ret = adxl375_set_odr_and_lp(dev, cfg->odr, cfg->lp);
+	if (ret < 0) {
+		LOG_ERR("Failed to set ODR and LP mode");
+		return ret;
+	}
+
+	// Need to force bits 3, 1 and 0 to set since the register defaults to 0
+	// See datasheet pages 20 and 24
+	ret = adxl375_set_data_format(dev, 0b00001011);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize data format");
+		return ret;
+	}
+
+	ret = adxl375_set_op_mode(dev, ADXL375_MEASUREMENT);
+	if (ret < 0) {
+		LOG_ERR("Failed to put in standby mode");
+	}
+
+	return ret;
+}
+
+static int adxl375_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct adxl375_data *data = dev->data;
+	uint8_t buff[6] = {0};
+
+	int ret = data->hw_tf->read_reg_multiple(dev, ADXL375_DATAX0, buff, 6);
+
+	data->sample.x = (int16_t)((buff[1] << 8) | (buff[0]));
+	data->sample.y = (int16_t)((buff[3] << 8) | (buff[2]));
+	data->sample.z = (int16_t)((buff[5] << 8) | (buff[4]));
+
+	return ret;
+}
+
+static void adxl375_accel_convert(struct sensor_value *result, int16_t sample_val)
+{
+	result->val1 = (sample_val * SENSOR_G * ADXL375_MG2G_MULTIPLIER) / 1000000;
+	result->val2 = ((int16_t)(sample_val * SENSOR_G * ADXL375_MG2G_MULTIPLIER)) % 1000000;
```

Comment:
This cast will truncate the value.
I suggest creating an intermediate scaled value, e.g.:
```suggestion
	int64_t value = (int64_t)sample_val * SENSOR_G * ADXL375_MG2G_MULTIPLIER;

	result->val1 = value / 1000000;
	result->val2 = value % 1000000;
```

Reply:
Ooh very nice catch. Did not see that. Thank you